src/components/Treeview/TreeviewItem.tsx

```diff
@@ -52,38 +72,55 @@ export class TreeviewItem extends CommonComponent<ITreeviewItemProps, any> {
 
         return (
             <div onMouseEnter={this._onItemHover} onMouseLeave={this._onItemLeaveHover} className={parentItemClassName}>
-                <div className={'treeview-item'}>
-                    {item.children.length > 0 &&
-                        <Icon iconName={this.state.iconArrow} onClick={this._onItemClick.bind(this)}></Icon>
+                <div className={classNames('treeview-item', item.className)}>
+                    {
+                        item.children.length > 0 &&
+                        <Icon iconName={arrowIcon} onClick={(ev) => this._onExpand(ev, item)}></Icon>
```

Comment:
try to avoid arrow functions. The item parameter is unnecessary since you can get it in the _onExpand method from state. You can then replace it with onClick={this._onExpand}